public/components/notebooks/components/__tests__/note_table.test.tsx

```diff
@@ -23,109 +24,261 @@ jest.mock('react-router-dom', () => ({
+    expect(props.cloneNotebook).toHaveBeenCalledTimes(1);
+    expect(props.cloneNotebook).toHaveBeenCalledWith('new-copy', 'id-1');
+  });
+
+  it('deletes a notebook', () => {
+    const notebooks = [
+      {
+        path: 'path-1',
+        id: 'id-1',
+        dateCreated: 'date-created',
+        dateModified: 'date-modified',
+      },
+    ];
+    const { getByText, getByLabelText, getAllByText, getByTestId } = renderNoteTable({ notebooks });
+
+    // Select a notebook
+    fireEvent.click(getByLabelText('Select this row'));
+
+    // Open Actions dropdown and click Delete
+    fireEvent.click(getByText('Actions'));
+    fireEvent.click(getByText('Delete'));
+
+    // Ensure the modal is open (you may need to adjust based on your modal implementation)
+    expect(getAllByText('Delete 1 notebook')).toHaveLength(1);
+
+    // Mock user confirmation and submit
+    fireEvent.input(getByTestId('delete-notebook-modal-input'), {
+      target: { value: 'delete' },
+    });
+    fireEvent.click(getByTestId('delete-notebook-modal-delete-button'));
+
+    // Assert that the deleteNotebook function is called
+    expect(props.deleteNotebook).toHaveBeenCalledTimes(1);
+    expect(props.deleteNotebook).toHaveBeenCalledWith(['id-1'], expect.any(String));
+  });
+
+  it('adds sample notebooks', async () => {
+    const { getByText, getAllByText, getByTestId } = renderNoteTable({ notebooks: [] });
+
+    // Open Actions dropdown and click Add samples
+    fireEvent.click(getByText('Actions'));
+    fireEvent.click(getAllByText('Add samples')[0]);
+
+    // Ensure the modal is open (you may need to adjust based on your modal implementation)
+    expect(getAllByText('Add sample notebooks')).toHaveLength(1);
+
+    // Mock user confirmation and submit
+    fireEvent.click(getByTestId('confirmModalConfirmButton'));
+
+    // Assert that the addSampleNotebooks function is called
+    expect(props.addSampleNotebooks).toHaveBeenCalledTimes(1);
+  });
+
+  it('closes the action panel', async () => {
+    const { getByText, queryByTestId } = renderNoteTable({ notebooks: [] });
+    expect(queryByTestId('rename-notebook-btn')).not.toBeInTheDocument();
+
+    // Open Actions dropdown
+    fireEvent.click(getByText('Actions'));
+
+    // Ensure the action panel is open
+    expect(queryByTestId('rename-notebook-btn')).toBeInTheDocument();
+
+    await act(async () => {
+      fireEvent.click(getByText('Actions'));
+    });
+
+    // Ensure the action panel is closed
+    expect(queryByTestId('rename-notebook-btn')).not.toBeInTheDocument();
+  });
+
+  it('closes the delete modal', () => {
+    const notebooks = [
+      {
+        path: 'path-1',
+        id: 'id-1',
+        dateCreated: 'date-created',
+        dateModified: 'date-modified',
+      },
+    ];
+    const { getByText, getByLabelText, queryByText } = renderNoteTable({ notebooks });
+
+    // Select a notebook
+    fireEvent.click(getByLabelText('Select this row'));
```

Comment:
There's a lot of duplicated navigation logic here based on label text, that seems like something that's likely to break over time. Should there be a helper for this navigation, or a better selector?

Reply:
I'll leave this one as is for now. I'm updating some of these with cypress.